06-building-a-program-start-to-finish/writing-tests/output/invertedIndexSpec.js

```diff
@@ -0,0 +1,161 @@
+    this.invertedIndex = new InvertedIndex();
+  });
+
+  describe('Constructor', () => {
+    it('can create inverted index instance', () => {
+      expect(typeof this.invertedIndex)
+        .toEqual('object');
+      expect(this.invertedIndex instanceof InvertedIndex)
+        .toBe(true);
+    });
+
+    it('has an indexes object to hold all indexes', () => {
+      expect(typeof this.invertedIndex.tableObj)
+        .toEqual('object');
+    });
+  });
+
+  describe('tokenize', () => {
+    it('checks the tokenize function', () => {
+      const input = 'jed is a boy'
+      const output = ['jed', 'is', 'a', 'boy'];
+      expect(this.invertedIndex.tokenize(input))
+        .toEqual(output);
+    });
+
+    it('sanitizes the input in the tokenize function', () => {
+      const input = 'tracy!!! is !@#$%^&*not -invited. &to my (wedding)'
+      const output = ['tracy', 'is', 'not', 'invited', 'to', 'my', 'wedding'];
+      expect(this.invertedIndex.tokenize(input))
+        .toEqual(output);
+    });
+  });
+
+  describe('getTextFromJsonObj', () => {
+    it('should return an array of words', () => {
+      expect(this.invertedIndex.getTextFromJsonObj(smallfile))
+        .toEqual(['alice', 'falls', 'into', 'a', 'rabbit']);
+    });
+
+    it('filters out symbols', () => {
+      expect(this.invertedIndex.getTextFromJsonObj(smallfile))
+        .toEqual(['alice', 'falls', 'into', 'a', 'rabbit']);
+    });
+  });
+
+  describe('uniqueWords', () => {
+    it('checks the uniqueWords function', () => {
+      const input = ['aa', 'aa', 'aa'];
+      const output = ['aa'];
+      expect(this.invertedIndex.uniqueWords(input))
+        .toEqual(output);
+    });
+  });
+
+  describe('Validate Files ', () => {
+    it('verifies that the JSON file is valid or invalid', () => {
+      expect(this.invertedIndex.validateFile(books)
+          .valid)
+        .toBe(true);
+      expect(this.invertedIndex.validateFile(smallfile)
+          .valid)
+        .toBe(true);
+      expect(this.invertedIndex.validateFile(emptyfile)
+          .valid)
+        .toBe(false);
+      expect(this.invertedIndex.validateFile(wrongdata)
+          .valid)
+        .toBe(false);
+    });
+  });
+
+  describe('CreateIndex', () => {
+    beforeAll(() => {
+      this.invertedIndex.createIndex(smallfile, ['alice', 'falls'], 'smallcorrectfile.json');
+      this.invertedIndex.createIndex(books, uniqueTermsBook, 'correct.json');
+    });
+    it('creates an index', () => {
+      expect(this.invertedIndex.getIndex('correct.json'))
+        .toBeTruthy();
+      expect(this.invertedIndex.getIndex('smallcorrectfile.json'))
+        .toBeTruthy();
+    });
+    it('creates the correct index', () => {
+      expect(this.invertedIndex.getIndex('correct.json')
+          .a)
+        .toEqual([true, true, true, true]);
+      expect(this.invertedIndex.getIndex('smallcorrectfile.json')
+          .falls)
+        .toEqual([true]);
+    });
+  });
+
+  describe('GetIndex', () => {
```

Comment:
You're testing your `getIndex` function here, while you've already used it in the `Create Index` suite. Are you aware your tests execute sequentially?